src/Features/Core/Portable/ConvertCast/AbstractConvertCastCodeRefactoringProvider.cs

```diff
@@ -54,7 +53,7 @@ public sealed override async Task ComputeRefactoringsAsync(CodeRefactoringContex
                 context.RegisterRefactoring(
                     CodeAction.Create(
                         title,
-                        c => ConvertAsync(document, from, cancellationToken),
+                        c => ConvertAsync(document, from, semanticModel, cancellationToken),
```

Comment:
dont' capture semanticModel here, it will keep it alive perpetually (which will hold onto the compilation).  reacquire from teh document in ConvertAsync.